src/ui/public/react_components.js

```diff
@@ -2,10 +2,15 @@ import 'ngreact';
 
 import {
   KuiToolBarSearchBox,
+  KuiConfirmModal,
 } from 'ui_framework/components';
 
 import { uiModules } from 'ui/modules';
+
 const app = uiModules.get('app/kibana', ['react']);
 app.directive('toolBarSearchBox', function (reactDirective) {
   return reactDirective(KuiToolBarSearchBox);
 });
+app.directive('confirmModal', function (reactDirective) {
```

Comment:
It gets imported in `public/kibana`, so anyone can use them.  For ui_framework components, which are generic, it's nice to include them once so you can use them everywhere.  For react components that are specific to a certain area, then I would import them where you need them.

So no, no plans to remove this.